jsonpatcher/jsonpatcher.go

```diff
@@ -239,20 +238,36 @@ func deleteEvent(id core.InstanceID) (*operation, error) {
 }
 
 type patchEvent struct {
-	Timestamp      int64
+	Timestamp      interface{}
 	ID             core.InstanceID
 	CollectionName string
 	Patch          operation
 }
 
 func (je patchEvent) Time() []byte {
-	t := je.Timestamp
+	var nanos int64
+	switch ts := je.Timestamp.(type) {
+	case time.Time:
+		nanos = ts.UnixNano()
+	case int64:
+		nanos = ts
+	}
 	buf := new(bytes.Buffer)
 	// Use big endian to preserve lexicographic sorting
-	_ = binary.Write(buf, binary.BigEndian, t)
+	_ = binary.Write(buf, binary.BigEndian, nanos)
 	return buf.Bytes()
 }
 
+func (je patchEvent) time() (t time.Time) {
+	switch ts := je.Timestamp.(type) {
+	case time.Time:
+		t = ts
+	case int:
```

Comment:
Should `s/int/int64` considering symmetry with `Time()` switch case?

Reply:
I did that initially, but the type actually comes back as `int` when it's set on an empty interface value.